### PR TITLE
Fix: invalid tokens

### PR DESCRIPTION
**Pull request by cocoshe**
## Related Issues:
+ #56 
+ #49 
+ #52 
+ #50 

### [Qwen/Qwen2.5-7B-Instruct](https://huggingface.co/Qwen/Qwen2.5-7B-Instruct/blob/main/generation_config.json#L11-L12)
+ https://huggingface.co/Qwen/Qwen2.5-7B-Instruct/blob/main/generation_config.json#L11-L12
```json
{
"bos_token_id": 151643,
"pad_token_id": 151643,
"do_sample": true,
"eos_token_id": [
151645,
151643
],
"repetition_penalty": 1.05,
"temperature": 0.7,
"top_p": 0.8,
"top_k": 20,
"transformers_version": "4.37.0"
}
```

### [Qwen/Qwen3-1.7B-Base](https://huggingface.co/Qwen/Qwen3-1.7B-Base/blob/main/generation_config.json)
```json
{
"bos_token_id": 151643,
"do_sample": false,
"eos_token_id": 151643,
"max_new_tokens": 2048,
"transformers_version": "4.37.0"
}
```

### [kkknight/MiniOneRec](https://huggingface.co/kkknight/MiniOneRec/blob/main/Industrial_ckpt/generation_config.json)

```json
{
"_from_model_config": true,
"bos_token_id": 151643,
"eos_token_id": 151645,
"transformers_version": "4.55.4"
}
```

`Instruct` model contain `top_k` and `top_p`, which lead to `TopKLogitsWarper`...
https://github.com/huggingface/transformers/blob/5a67f0a7aca8c18de6235e80e47fd00a9ecf9d58/src/transformers/generation/logits_process.py#L581-L586

may filter the 50 valid tokens(after constrained beam search=`50`) down to even `only 2` valid tokens(valid means logits is not `-inf`). So the other 50-2=48 breams are selected in all the tokens with logits == `-inf` with `top-k` beam selection.

ce54caa3698d472a5a4b10ac7d9fdb6b

#### Since so many related issues, I highly recommend to fix this to support `instrct` model to align with the model in paper.

> Qwen2.5-1.5B-instrct sft result before fixing the bug
f1ea6eaa66027a742dbc7fbfdda96904

> Qwen2.5-1.5B-instrct stf result after fixing the bug
a34b4a1b5897d59fd053cbbc2aac525b

### CODE REVIEW

**Comment by hrdxwandg** [comment]
Qwen2.5-1.7B-instrct？有这个模型么？
我更新代码后，用Qwen2.5-3B-Instruct sft 后还是复现不了，我用的 repo 自带的数据，我的结果如下：
image

**Comment by cocoshe** [comment]
> Qwen2.5-1.7B-instrct？有这个模型么？ 我更新代码后，用Qwen2.5-3B-Instruct sft 后还是复现不了，我用的 repo 自带的数据，我的结果如下： image

Sorry for the typo, it should be qwen2.5-1.5B-instruct.

I think it is somehow difficult to reproduce, it requires the author to provide detailed experiment log to carefully set the hyperparameters. I still suffer from it
